packages/utils/core-utils/src/package-manager.ts

```diff
@@ -0,0 +1,44 @@
+import { CrossSpawnArgs, CrossSpawnOptions, spawn } from '@malept/cross-spawn-promise';
+import chalk from 'chalk';
+import logSymbols from 'log-symbols';
+import yarnOrNpm from 'yarn-or-npm';
+
+export type PackageManager = 'npm' | 'yarn' | 'pnpm';
+
+export const getPackageManager = (): PackageManager => {
+  const userAgent = process.env.npm_config_user_agent || '';
+  const system = yarnOrNpm();
+
+  if (userAgent.startsWith('yarn')) {
+    return 'yarn';
+  }
+
+  if (userAgent.startsWith('npm')) {
+    return 'npm';
+  }
+
+  if (userAgent.startsWith('pnpm')) {
+    return 'pnpm';
+  }
+
+  // use NODE_INSTALLER as an override of npm_config_user_agent
+  switch (process.env.NODE_INSTALLER) {
+    case 'yarn':
+    case 'npm':
+    case 'pnpm':
+      return process.env.NODE_INSTALLER;
+    default:
+      if (process.env.NODE_INSTALLER) {
+        console.warn(logSymbols.warning, chalk.yellow(`Unknown NODE_INSTALLER, using detected installer ${system}`));
```

Comment:
I suggest using [detect-package-manager](https://www.npmjs.com/package/detect-package-manager) instead of yarn-or-npm. Because `detect-package-manager` supports pnpm.

```suggestion
        console.warn(logSymbols.warning, chalk.yellow(`Not found NODE_INSTALLER env, using detected installer ${system}`));
```

Reply:
I think `Unknown` sounds more correct than `Not found`